deploy/examples/create-external-cluster-resources.py

```diff
@@ -540,35 +541,53 @@ def _check_conflicting_options(self):
             )
 
     def _invalid_endpoint(self, endpoint_str):
+        # seprating port, by getting last split of `:` delimiter
         try:
-            ipv4, port = endpoint_str.split(":")
+            endpoint_str_ip, port = endpoint_str.rsplit(":", 1)
         except ValueError:
-            raise ExecutionFailureException(
-                f"Not a proper endpoint: {endpoint_str}, <IPv4>:<PORT>, format is expected"
+            raise ExecutionFailureException(f"Not a proper endpoint: {endpoint_str}")
+
+        try:
+            if endpoint_str_ip[0] == "[":
+                # endpoint_str_ip = endpoint_str_ip[]
+            ip_type = (
+                "IPv4" if type(ip_address(endpoint_str_ip)) is IPv4Address else "IPv6"
```

Comment:
We may also use `IPv6Address` library to check if it is ipv6 ip, currently IIUC, we are checking if it is not ip4 then make it ipv6?